### PR TITLE
fix #442 and part of #443

### LINKED ISSUE #442: util/scripts/install-deps-ubuntu.sh broken on fresh installed Ubuntu 16.04
Ubuntu 16.04, fresh installed within VMWare.

```
qianyizh@ubuntu:~/Open3D$ sudo apt-get update
Get:1 http://security.ubuntu.com/ubuntu xenial-security InRelease [107 kB]
Hit:2 http://us.archive.ubuntu.com/ubuntu xenial InRelease
Get:3 http://us.archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]        
Get:4 http://us.archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]                
Fetched 323 kB in 0s (347 kB/s)                             
Reading package lists... Done

qianyizh@ubuntu:~/Open3D$ util/scripts/install-deps-ubuntu.sh 

sudo apt-get install \
		xorg-dev \
		libglu1-mesa-dev \
		libgl1-mesa-glx \
		libglew-dev \
		libglfw3-dev \
		libjsoncpp-dev \
		libeigen3-dev \
		libpng-dev \
		libpng16-dev \
		libjpeg-dev \
		pybind11-dev \
		python-dev \
		python3-dev \
		python-tk \
		python3-tk
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'libpng12-dev' instead of 'libpng-dev'
E: Unable to locate package pybind11-dev```

### PR DESCRIPTION
**Pull request by qianyizh**
fix #442 and part of #443

### CODE REVIEW

File: util/scripts/install-deps-ubuntu.sh [hunk 1]
```diff
@@ -11,10 +13,10 @@ sudo apt-get install \
...
 		libpng-dev \
-		libpng16-dev \
 		libjpeg-dev \
-		pybind11-dev \
 		python-dev \
 		python3-dev \
 		python-tk \
 		python3-tk
+
+sudo apt-get install libpng16-dev
```
Comment (qianyizh) [inline util/scripts/install-deps-ubuntu.sh]: Have to separate libpng-dev and libpng16-dev or there will be error.

**Comment by qianyizh** [comment]
Travis is on Ubuntu 14.04 so this is broken (glfw3 not found).
I thought about it and think the easiest solution is to split the dependencies into multiple apt-get commands (so any failure won't cause the others to fail). Then address #407 to auto detect dependencies.

**Comment by qianyizh** [comment]
Will fix it tonight.

File: .travis.yml [hunk 2]
```diff
@@ -75,12 +92,16 @@ matrix:
...
       osx_image: xcode8.3
 
+    - os: osx
+      osx_image: xcode8.3
+      env: BUILD_DEPENDENCY_FROM_SOURCE=ON
+
 before_install:
   # ISSUE: we may want to use '/scripts/install-deps-ubuntu.sh' in the future.
   # At this moment, travis-ci supports up to ubuntu 14.04 and this environment
   # does not recognize sudo apt-get install libglfw3-dev
```
Comment (syncle) [inline .travis.yml]: Now we can remove this comment?

Comment (qianyizh) [reply .travis.yml]: I am still debugging. And yes I want to get rid of it. I want to use util/scripts/install-deps-ubuntu.sh.

**Comment by qianyizh** [comment]
Once CI's finish, this PR should be ready to merge. @syncle 
This resolves #442 #443 #445 #407 and part of #366 on Visual Studio 2017.
Besides CI (OSX and Ubuntu 14.04), I tested manually on Windows Visual Studio 2015, Visual Studio 2017, Ubuntu 16.04, Ubuntu 18.04. All work. The compiling from source workflow should be good now.
I still need to do some polish on documents. This will be a followup PR.

**Pull request merged**